### ISSUE TITLE
Commits should be signed and author emails checked by CI

### ISSUE THREAD

**Issue by ttoine** [PrestaShop/PrestaShop#27557]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

There is currently no check on the identity of a commit author.
We also know that there are people using different emails on their local git configuration and with GitHub.
This makes tracking code provenance and managing intellectual property difficult.

### Describe the solution you'd like

Ask developers to:
- use the same email in their local git configuration and in GitHub
- sign their commit (maybe we should take this occasion to introduce a [DCO](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin)?)
- use the right email (profesionnal when paid to contribute, personal when contributing on their free time)

Then, on each PR, or each PR modification/improvement:
- CI must check that commits are signed
- CI must check that emails are the good ones 

### Alternatives you've considered

-

### Additional context

Most serious open source project do that from start. Knowing authors and checking that they don't use someone else identity is essential in intellectual property management of a distributed software project.

### Do you plan to work on this feature?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by marwachelly** [comment]
Hello, 

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know!

Thanks,

**Comment by marwachelly** [comment]
Hello,

Thank you for your suggestion. The @PrestaShop/prestashop-core-developers  will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know!

Thanks,

**Comment by kpodemski** [comment]
I completely agree with the fact that it might be in some way valuable for the project, but if, for example, Symfony doesn't force it, then why would PrestaShop need to force it from the community contributors?

Maybe let's try to start from the maintainers, but honestly, I don't see it as a requirement for community contributions. I remember how problematic it was for the community contributors to squash the commits - this is no longer a requirement. Imagine requiring someone to set up everything to have signed commits... :)

**Comment by ttoine** [comment]
having it for maintainers would be at least a good thing.

note that I continue my work on the intellectual property management of the PrestaShop project, so I list problems to solve and possible improvements. one of the problem we have currently is that we don't know authors.

btw, this is the kind of usual prerequisite to contribute to most projects managed by a foundation.
